File: Atcoder_#346/javascript/b_piano.js
const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

rl.on('line', (line) => {
  const [w, b] = line.split(' ').map(Number);
  console.log(existsWbSubstring(w, b));
  rl.close();
});

function existsWbSubstring(w, b) {
  const s = 'wbwbwwbwbwbw';
  const sSize = s.length;
  const targetSize = w + b;

  // 必要に応じて、十分な長さに文字列を拡張
  const repeatTimes = Math.ceil(targetSize / sSize) + 1;
  const extendedS = s.repeat(repeatTimes);

  // 部分文字列をスライドしながらチェック
  for (let i = 0; i <= extendedS.length - targetSize; i++) {
    const substring = extendedS.slice(i, i + targetSize);
    const wCount = (substring.match(/w/g) || []).length;
    const bCount = (substring.match(/b/g) || []).length;
    if (wCount === w && bCount === b) {
      return "Yes";
    }
  }

  return "No";
}

File: Atcoder_#346/ruby/b_piano.rb
def exists_w_b_substring?(w, b)
  s = 'wbwbwwbwbwbw'
  s_size = s.size
  target_size = w + b

  # 必要に応じて、十分な長さに文字列を拡張
  extended_s = s * ((target_size / s_size.to_f).ceil + 1)

  # 部分文字列をスライドしながらチェック
  0.upto(extended_s.size - target_size) do |i|
    substring = extended_s[i, target_size]
    w_count = substring.count('w')
    b_count = substring.count('b')
    if w_count == w && b_count == b
      return "Yes"
    end
  end

  "No"
end

# 入力を受け取る
w, b = gets.strip.split.map(&:to_i)

# 結果を出力
puts exists_w_b_substring?(w, b)
